controller/appcontroller.go

```diff
@@ -188,8 +214,25 @@ func (ctrl *ApplicationController) setAppManagedResources(a *appv1.Application,
 func (ctrl *ApplicationController) getResourceTree(a *appv1.Application, managedResources []*appv1.ResourceDiff) (*appv1.ApplicationTree, error) {
 	nodes := make([]appv1.ResourceNode, 0)
 
+	proj, err := argo.GetAppProject(&a.Spec, applisters.NewAppProjectLister(ctrl.projInformer.GetIndexer()), ctrl.namespace)
+	if err != nil {
+		return nil, err
+	}
+	orphanedNodesMap := make(map[kube.ResourceKey]appv1.ResourceNode)
+	warnOrphaned := true
+	if proj.Spec.OrphanedResources != nil {
+		orphanedNodesMap, err = ctrl.stateCache.GetNamespaceTopLevelResources(a.Spec.Destination.Server, a.Spec.Destination.Namespace)
+		if err != nil {
+			return nil, err
+		}
+		if proj.Spec.OrphanedResources.Warn != nil {
+			warnOrphaned = *proj.Spec.OrphanedResources.Warn
+		}
+	}
+
 	for i := range managedResources {
 		managedResource := managedResources[i]
+		delete(orphanedNodesMap, kube.NewResourceKey(managedResource.Group, managedResource.Kind, managedResource.Namespace, managedResource.Name))
```

Comment:
how about creating `managedResources.ResoureKey()`

Reply:
👍

Reply:
had to revert this change because this introduces dependency between `pkg/apis/application/v1alpha1` and `utils/kube`. It causes circular dependency in multiple packages :(